src/ui/viewer.rs

```diff
@@ -37,6 +37,24 @@ pub trait SnarlViewer<T> {
         let _ = (node, inputs, outputs, snarl);
         default
     }
+    /// Checks if node has a custom egui style.
+    #[inline]
+    fn has_node_style(&mut self, node: &T) -> bool {
```

Comment:
Should take node to consider style as other methods do.
```rust
node: NodeId,
inputs: &[InPin],
outputs: &[OutPin],
snarl: &Snarl<T>,
```

Reply:
I have used `has_body` as a reference point, which also only accepts the NodeId

Reply:
I understand. But I can imagine someone wishing to decide on style based on connected nodes.
`has_body` should also take all the info.

Reply:
@zakarumych I made the change for `has_node_style`. Similar change for other `has_*` should be done, but out of scope of this PR.